src/libs/deepReplaceKeysAndValues.ts

```diff
@@ -0,0 +1,53 @@
+type ReplaceableValue = Record<string, unknown> | unknown[] | string | number | boolean | undefined | null;
+
+/**
+ * @param  obj the object to transform
+ * @param oldVal the value to search for
+ * @param newVal the replacement value
+ */
+function deepReplaceKeysAndValues<T extends ReplaceableValue>(obj: T, oldVal: string, newVal: string): T {
+    if (!obj) {
+        return obj;
+    }
+
+    if (typeof obj === 'string') {
+        return obj.replace(oldVal, newVal) as T;
+    }
+
+    if (typeof obj !== 'object') {
+        return obj;
+    }
+
+    if (Array.isArray(obj)) {
+        return obj.map((item) => deepReplaceKeysAndValues(item as T, oldVal, newVal)) as T;
+    }
+
+    const newObj: Record<string, unknown> = {};
+    Object.entries(obj).forEach(([key, val]) => {
+        let newKey = key === oldVal ? newVal : key;
+        if (typeof newKey === 'string') {
+            newKey = newKey.replace(oldVal, newVal);
+        }
```

Comment:
Keys are always strings

```suggestion
        let newKey = key.replace(oldVal, newVal);
```

Reply:
thanks, updated